utils/model_to_sql.py

```diff
@@ -0,0 +1,489 @@
+    if len(sys.argv) <= 1:
+        print(
+            "Usage: python hf_data_catalog <operation> <schema> <model_path> [<ro_db_users>] [<rw_db_users>]"
+        )
+        return -1
+    if len(sys.argv) <= 2:
+        print(
+            "Error: Must specify schema as either development, public or public_test in command line argument."
+        )
+        return -1
+    if len(sys.argv) <= 3:
+        print("Error: Must specify the model path in the command line argument.")
+        return -1
+
+    operation = sys.argv[1]
+    schema = sys.argv[2]
+    model_path = sys.argv[3]
+    ro_users_str = sys.argv[4] if len(sys.argv) > 4 else None
+    rw_users_str = sys.argv[5] if len(sys.argv) > 5 else None
+
+    if len(sys.argv) > 6:
+        print("Error: Too many arguments specified")
+        return -1
+
+    # Allow the optional list of ro_users and rw_users to have a default value
+    ro_users = (
+        ro_users_str.split(",")
+        if ro_users_str is not None
+        else ["hmei-hydro", "data_catalog-rw", "data_catalog-ro"]
+    )
+    rw_users = rw_users_str.split(",") if rw_users_str is not None else ["hmei-hydro"]
+
+    if operation not in ["drop", "import", "export"]:
+        print("Error: The operation must be drop, import or export")
+        return -1
+    if schema not in ["development", "public_test", "public"]:
+        print("Error: The schema must be development, public or public_test")
+        return -1
+    if not os.path.exists(model_path):
+        print(f"Model path '${model_path}' does not exist.")
+
+    if operation == "drop":
+        drop_all_tables(model_path, schema)
+    elif operation == "import":
+        import_to_sql(model_path, schema, ro_users, rw_users)
+    elif operation == "export":
+        export_to_csv(model_path, schema)
+    return 0
+
+
+def export_to_csv(model_path: str, schema: str = "development"):
+    """Export the model from the SQL schema into the model_path directory as csv files for each table."""
+
+    if not os.path.exists(model_path):
+        raise ValueError(f"The folder '{model_path}' does not exist.")
+    connection = get_connection(schema)
+    with connection:
+        _add_current_date_version(connection)
+        table_names = _get_list_of_tables(connection, schema)
+        for table_name in table_names:
+            column_names = _get_columns_of_table(connection, schema, table_name)
+            rows = _get_row_values_from_table(connection, table_name)
+            with open(f"{model_path}/{table_name}.csv", "w+") as stream:
+                header = ",".join(column_names)
+                stream.write(f"{header}\n")
+                for row in rows:
+                    data_values = [
+                        _format_export(row.get(column_name), column_name)
+                        for column_name in column_names
+                    ]
+                    data_row = ",".join(data_values)
+                    stream.write(f"{data_row}\n")
+
+
+def import_to_sql(
```

Comment:
Does this mean that the main way of updating the database is by writing a csv and importing that?

Reply:
This import_to_sql function is what I'm using to import the existing .csv files to import into the database. We still need it for migration.

I kept this code because I'm currently exporting the database to .csv files into an archive folder after we publish to the public schema. By keeping the import function around I could import back one of those archive .csv folders. This is "probably" not necessary after we migrate to use the SQL database, but it doesn't hurt to keep it for a while just in case. In my old job we always did DB backups as .csv because they are technology independent backups. Our new data catalog workflow does not require or use this, but it makes me feel safer about not losing anything.

Reply:
That makes a lot of sense. Thanks!